src/main/java/com/google/sps/algorithms/MatchRanking.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.algorithms;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class MatchRanking {
+
+    /**
+    * Function takes collection of user's saved bios, all user bios, and new match bios and 
+    * returns a list of the new match bios in ranked order from highest to lowest score 
+    * (higher score = more likely that the user will select it). 
+    */
+    public static List<String> query(Collection<String> savedMatchBios, Collection<String> allUserBios, Collection<String> newMatchBios) {
```

Comment:
nit: maybe call this rankMatches or rankBios as query sounds a bit ambiguous.

Reply:
Done!